src/coreclr/jit/morph.cpp

```diff
@@ -14290,6 +14290,55 @@ GenTree* Compiler::fgMorphSmpOpOptional(GenTreeOp* tree)
             }
             break;
 
+#if defined(FEATURE_HW_INTRINSICS) && defined(TARGET_XARCH)
+        case GT_HWINTRINSIC:
+        {
+            GenTreeHWIntrinsic* hw = tree->AsHWIntrinsic();
+            switch (hw->gtHWIntrinsicId)
+            {
+                case NI_SSE_Xor:
+                case NI_SSE2_Xor:
+                case NI_AVX_Xor:
+                case NI_AVX2_Xor:
+                {
+                    // Optimize Xor(x, Vector_.Zero) to just x
+                    GenTree* op1 = hw->gtGetOp1();
+                    GenTree* op2 = hw->gtGetOp2();
+
+                    // Is node - Vector_.Zero ?
+                    auto isHwZero = [](GenTree* node) -> bool {
```

Comment:
I thought we had a helper function that checks if its any type of SIMD zero?

Reply:
`IsIntegralConstVector(0)`: https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/gentree.h#L7252

Reply:
Doesn't that work only for integer types? This code seems to also operate on floating point zero since it works for NI_SSE_Xor and that is only for floating point data.

Reply:
If we want to include floating-point we should likely have a new general helper method that accounts for any vector zero kind and that can easily be expanded to include other zeroes if that's ever important.

Reply:
It's fine to handle only integer types for now, this patterns shows up after Vector.Equals only for integers.